### PR TITLE
Issue 7048

### PR DESCRIPTION
**Pull request by mattmemmesheimer**
This is my first attempt at contributing to CakePHP, so I hope I did everything correctly.

To address, issue #7048,  I changed the bake utility to use FlashComponent::success or FlashComponent::error instead of the deprecated SessionComponent::setFlash.

### CODE REVIEW

**Comment by dereuromark** [comment]
Looks pretty good to me.
You have a chmod change though that is not neither intential nor good, you might want to revert that.

**Comment by mattmemmesheimer** [comment]
Ah yes I'm not sure how that happened, my mistake.  I have reverted it.

File: lib/Cake/View/Elements/Flash/default.ctp [hunk 1]
```diff
@@ -1 +1,7 @@
-<div id="<?php echo $key; ?>Message" class="<?php echo !empty($params['class']) ? $params['class'] : 'message'; ?>"><?php echo $message; ?></div>
\ No newline at end of file
+<?php
+$class = 'message';
+if (!empty($params['class'])) {
+    $class .= ' ' . $params['class'];
+}
+?>
+<div class="<?= h($class) ?>"><?= h($message) ?></div>
```
Comment (markstory) [inline lib/Cake/View/Elements/Flash/default.ctp]: The lack of id attribute may cause applications to be styled incorrectly when they upgrade. Can we retain the id attribute for now?

**Comment by markstory** [comment]
Other than the loss of the id attribute, this looks good to me.

**Comment by mattmemmesheimer** [comment]
Should the call to `$this->Session->flash()` be replaced with `$this->Flash->render()` in the default layout as well? If so, we can change the FlashHelper to include the `$key` in the view variables so that each flash template file can include `id="<?= $key ?>Message"`.

**Comment by markstory** [comment]
@mattmemmesheimer I think it makes sense to update the views to use the new helper as well.

**Comment by markstory** [comment]
Merging and I'll fix the short tags post merge.

**Pull request merged**